sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business.service.ts

```diff
@@ -0,0 +1,128 @@
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationSINData = await this.getSINForApplications(
+      applicationNumbers,
+      institutionId,
+    );
+    applicationSINData.forEach((record) => {
+      applicationSINDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
+      };
+    });
+    const businessValidationModels: ApplicationBulkWithdrawalImportBusinessValidationModel[] =
+      [];
+    textValidations.forEach((textValidation) => {
+      const businessValidationModel =
+        {} as ApplicationBulkWithdrawalImportBusinessValidationModel;
+      const applicationData =
+        applicationSINDataMap[textValidation.textModel.applicationNumber];
+      if (applicationData) {
+        businessValidationModel["applicationFound"] = true;
+        businessValidationModel["applicationBelongsToInstitution"] =
+          allowedLocationIds.includes(applicationData.locationId);
+        businessValidationModel["validSIN"] =
+          textValidation.textModel.sin === applicationData.sin;
+        businessValidationModel["hasCorrectInstitutionCode"] =
+          textHeader.originator === applicationData.locationCode;
+        businessValidationModel["isArchived"] = applicationData.isArchived;
+        businessValidationModel["applicationStatus"] =
+          applicationData.applicationStatus;
+        businessValidationModel["hasPreviouslyBeenWithdrawn"] =
+          applicationData.hasPreviouslyBeenWithdrawn;
+        businessValidationModel["isRecordMatch"] =
+          businessValidationModel["applicationBelongsToInstitution"] &&
+          businessValidationModel["validSIN"];
+      } else {
+        businessValidationModel["applicationFound"] = false;
+      }
+      businessValidationModel["recordType"] =
+        textValidation.textModel.recordType;
+      businessValidationModel["sin"] = textValidation.textModel.sin;
+      businessValidationModel["applicationNumber"] =
+        textValidation.textModel.applicationNumber;
+      businessValidationModel["withdrawalDate"] =
+        textValidation.textModel.withdrawalDate;
+      businessValidationModels.push(businessValidationModel);
+    });
+    return businessValidationModels;
+  }
+
+  /**
+   * Get the SIN and location information for each of the provided applications.
+   * @param applicationNumbers application numbers for which the SIN and location information needs to be retrieved.
+   * @param institutionId institution id.
+   * @returns applications containing the required information.
+   */
+  async getSINForApplications(
```

Comment:
If this method is not intended to be shared by other services it can be marked as private. maybe the mapping part can happen here or is another private method to reduce the amount of line in `generateBusinessValidationModels`.